src/Components/SlateEditor/SlateEditor.js

```diff
@@ -124,6 +172,87 @@ export default function SlateEditor() {
     },
   };
 
+  const getCoordinates = () => {
+    let x, y;
+    const selection = window.getSelection();
+    if (selection.rangeCount !== 0) {
+      const range = selection.getRangeAt(0).cloneRange();
+      range.collapse(false);
+      const rect = range.getClientRects()[0];
+      if (rect) {
+        x = rect.top;
+        y = rect.left;
+      }
+      return { top: x > 120 ? x - 150 : x + 20, left: y };
+    } else {
+      alert("Internal server error, please try again later");
+    }
+  };
+
+  const closeMenu = useCallback(() => {
+    setShowMenu(false);
+    document.removeEventListener("click", closeMenu);
+  }, [setShowMenu]);
+
+  const openMenu = useCallback(() => {
+    const position = getCoordinates();
+    setCoordinates(position);
+    setShowMenu(true);
+    document.addEventListener("click", closeMenu);
+  }, [closeMenu, setCoordinates, setShowMenu]);
+
+  const editorOnKeyDown = useCallback(
+    (event) => {
+      if (
+        event.key === " " ||
+        event.key === "Backspace" ||
+        event.key === "Escape"
+      ) {
+        closeMenu();
+        ReactEditor.focus(editor);
+        event.preventDefault();
+      }
+
+      if (event.key === "Enter") {
+        event.preventDefault();
+      }
+    },
+    [closeMenu, editor]
+  );
+
+  const markdownListMenuOptions = menuOptions.map((item, key) => {
+    return (
+      <MenuItem
+        tabIndex="0"
+        key={key}
+        onClick={() => {
```

Comment:
I thought I added a comment here about moving to a separate function with useCallback here? 
If not, it will help improve performance by doing that